File: S2_07052025/Ex2_7.m
% multiplying two or more image
clear, clc, close all;
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
pkg load image; %load image package in Octave

%Load test image
I1 = im2double(imread('earth1.jpg'));
I2 = im2double(imread('earth2.jpg'));
I = immultiply(immultiply(immultiply(I1, I2), I1), I2);

%plot image
FS = 15
figure, clf;

subplot(1, 2, 1), imshow(I1, [0, 1]), title('earth 2D Image', "fontsize", FS);
subplot(1, 2, 2), imshow(I, [0, 1]), title('earth 3D Image', "fontsize", FS);

saveas(gcf, 'earth_multiply.jpg');

File: S2_07052025/ConvertPython/Ex2_Python/Ex2_7.py
import cv2
import numpy as np
import matplotlib.pyplot as plt

I1 = cv2.imread('earth1.jpg') / 255.0
I2 = cv2.imread('earth2.jpg') / 255.0
I = I1 * I2 * I1 * I2
plt.figure(figsize=(8, 4))
plt.subplot(1, 2, 1), plt.imshow(I1), plt.title('earth 2D Image', fontsize=15), plt.axis('off')
plt.subplot(1, 2, 2), plt.imshow(I), plt.title('earth 3D Image', fontsize=15), plt.axis('off')
plt.show()
plt.savefig('earth_multiply.jpg')